### PR TITLE
Prevent scheduler hanging

### PR DESCRIPTION
**Pull request by sfroemkenjw**
In the meanwhile our customer has over 10.000 files in root directory. To prevent memory_limit exceed I have added a yield generator.
As all files are synced with help of scheduler task, there is no need to test existing or missing files against bynder API again. So, I have modified the getFile API call. Please set second argument to true, to get the real answer from Bynder API, but EXT:bynder2 itself, will not use that anymore.
TYPO3's file indexer will only mark files as deleted, but will not delete the file itself. This is correct behavior, BUT after calculating the DIFF, EXT:bynder2 starts over 1000 requests to Bynder2 API (as customer has deleted as many files within 3 years) to test if file was "realy" deleted. With version 2.0.2 we prevent that additional call, as we just have retrieved a fresh list of file and file information from Bynder API in scheduler task.

### CODE REVIEW

File: Classes/Command/SyncBynderFilesCommand.php [hunk 1]
```diff
@@ -50,55 +94,55 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
-        // Check and mark missing files
-        $this->getIndexer($storage)->processChangesInStorages();
+        // This will retrieve ALL files from Bynder API.
+        // The fileInfo caches will be updated,
+        // detect changes in storage,
+        // check and mark missing files
+        try {
+            $this->getIndexer($bynderStorage)->processChangesInStorages();
+        } catch (\Exception $e) {
+            $this->logger->error('TYPO3 indexer error: ' . $e->getMessage());
```
Comment (hojalatheef) [inline Classes/Command/SyncBynderFilesCommand.php:135]: It's better to share little more info like the exception class file and line where the exception occured. Isn't it?
Share your thoughts about that.

`$this->logger->error('TYPO3 indexer error: ' . $e->getMessage(), [
    'exception' => $e,
]);
`

File: Classes/Service/BynderService.php [hunk 2]
```diff
@@ -56,21 +61,35 @@ public function __construct(array $configuration)
...
+        }
+    }
 
-        $this->extConf = GeneralUtility::makeInstance(ExtConf::class);
+    /**
+     * Because constructor will be called with arguments we inject FileInfo Cache with DI
+     */
+    public function setFileInfoCache(FrontendInterface $fileInfoCache)
+    {
+        $this->fileInfoCache = $fileInfoCache;
```
Comment (hojalatheef) [inline Classes/Service/BynderService.php:92]: property declared dynamically and it's not defined in the class. Make sure its's defined in the class.

**Review by hojalatheef — COMMENTED** [review]
Overall the implementation looks promising for handling the issue usecase. Just some comments I mentioned regarding the error reporting with context for better debugging and one property defined dynamically inside a class. Otherwise everything looks fine 👌

File: Classes/Command/SyncBynderFilesCommand.php (hunk 1 above)
Comment (sfroemkenjw) [reply Classes/Command/SyncBynderFilesCommand.php:135]: Good point. Done

**Pull request merged**